### ISSUE TITLE
DS UI filtering on Page Type for Navigation buggy

### ISSUE THREAD

**Issue by cylkee** [symphonycms/symphony-2#2649]
I've noticed that when filtering on a Page Type, it only works for root subjects, e.g.

Root page `/foo` has Page Type "Foo". Its child page `/foo/bar` has an empty or other Page Type.  When using "Foo" as the filter, the child page will be included in the XML no matter its Page Type.

```
<navigation>
    <page handle="foo" id="1">
        <name>Foo</name>
        <types>
            <type>Foo</type>
        </types>
        <page handle="bar" id="2">
            <name>Bar</name>
        </page>
    </page>
</navigation>
```

If I set a secondary filter of '/foo' on Parent Page, of the two pages in this example the <navigation> will be empty when the child page does not match the filter.

Suppose I filter on '/foo' for Parent Page which has three child pages, only two of which have the Page Type of "Foo", if I filter on that, too, only two of the child pages are output in `<navigation>`.  This seems to be the desired result, but only because they are now output as the first level.

Feature or bug? :)

**Comment by nitriques** [comment]
> Feature or bug? :)

@michael-e @brendo ? I would say bug.... but I am not sure. All pages should be filtered ?

**Comment by michael-e** [comment]
I don't really understand the description of the issue, and I am afraid that we need a very systematical approach here, probably combining testing and reverse engineering.

**Comment by nitriques** [comment]
Meaning if @cylkee is right this is what he gets when filtering on page types: First level are properly filtered but not their children. All children are always returned.

**Comment by cylkee** [comment]
> When using "Foo" as the filter, the child page will be included in the XML no matter its Page Type.

> First level are properly filtered but not their children. All children are always returned.

Correct.  I came across this while trying to create a Navigation data source in a Symphony `2.6.11` installation in which I have created multiple Pages of two main Types and none, with both main Types of Pages having child Pages.
